### PR TITLE
chore: upgrade to ReasonRelay 0.5.2 and add custom scalars

### PR DESCRIPTION
**Pull request by zth**
Hey!

Hope it's fine I'm doing this, it's just so nice to see someone else's code using ReasonRelay, and since I keep breaking APIs I figured it'd be nice of me to actually fix it too :sweat_smile: 

This PR upgrades to ReasonRelay `0.5.2`, which (yet again) change how fragments work, by removing the weird unwrap API in favor of adding a `getFragmentRefs()` function to each record containing fragment references. This should hopefully be a lot easier and mean less mental overhead..

I also added `Datetime` and `Cursor` to custom scalars in `relay.config.js`, so you won't have to manually cast them from `any` to `string`. I realize custom scalars aren't mentioned anywhere in the docs - I gotta fix that too eventually :sweat_smile: 

Anyway, do what you want with this PR. And as usual, any feedback on the APIs etc is much welcome!

### CODE REVIEW

**Comment by believer** [comment]
@zth Wow, this is amazing, thank you so much! 😄 I didn't know you could use `customScalars`, I'm quite new to Relay and couldn't find anything about such a configuration in their docs 😄 

Maybe you know if I'm doing something odd in `module SoundtrackEnum` in `AddSoundtrack.re`. I couldn't find a way to wrap the `MOVIE` polyvariant to the string `"MOVIE"` when passing it in a mutation? The mutation input expects     `soundtrackType: SchemaAssets.Enum_SoundtrackType.t`, but the actual value you should send to the API is a string.

**Comment by zth** [comment]
Yeah, the docs are sadly pretty bad still... :( I think I'll try and cover as much of Relay as I can in the ReasonRelay docs, so things like this don't get lost :sweat_smile: 

Re `SoundTrackEnum` - all enums should automatically be converted to strings before being sent to the API, so if that's not happening here then it's a bug. You're doing nothing wrong, and you can safely remove the code you have for stringifying etc. _If you really need_ to turn an enum into a string, there's now a `toString` method for the enum in `SchemaAssets`. 

To summarize - this looks like a bug, I'll have a deeper look at it!

**Comment by believer** [comment]
Oh I wasn't talking about your docs, I couldn't find it in the Relay docs 😅 

Ahh cool, I thought that might be the case since it's automatically converted when querying! I was getting a GraphQL error when sending the mutation that the `soundtrackType` can't be an `int` (what the polyvariant looks like in JS, like `12345678`) if that helps in any way 😄

**Comment by zth** [comment]
Yeah, I was talking about the Relay docs too. :/

I've found the issue and I'm fixing it! I'll get back to you here when it's done :sweat_smile:

**Comment by believer** [comment]
@zth that's great, thanks a bunch! 😄

**Comment by zth** [comment]
So, this is gonna take a little time to get right. A workaround until this is fixed for real could probably be something like sending the enum as: `yourEnumValue->SchemaAssets.Enum_yourEnum.toString->Obj.magic`. Horrible, but it'll likely let you continuenuntil I've fixed it :grin:

**Comment by believer** [comment]
Yeah, no worries! No stress what so ever, I appreciate all the work you're doing! haha good ol' `Obj.magic` at least makes the solution simpler for now 😄

**Comment by zth** [comment]
@believer quick report back - this uncovered all sorts of hairy stuff I've missed :sweat_smile: Working on fixing it and I'm pretty close. Just wanted to drop by and say thanks for finding this :grin:

**Comment by believer** [comment]
@zth ohh sorry about that! 😂 That's great though, looking forward to it! Thanks for all the hard work
